src/System.Private.CoreLib/shared/System/Decimal.DecCalc.cs

```diff
@@ -2514,22 +2503,22 @@ internal static void InternalRound(ref DecCalc d, uint scale, RoundingMode mode)
                     if (power >= remainder)
                         goto done;
                 }
-                else if (mode == RoundingMode.AwayFromZero)
+                else if (mode == MidpointRounding.AwayFromZero)
                 {
                     // Round away from zero at the mid point.
                     remainder <<= 1;
                     if (power > remainder)
                         goto done;
                 }
-                else if (mode == RoundingMode.Floor)
+                else if (mode == MidpointRounding.ToNegativeInfinity)
```

Comment:
:exclamation: `Floor` and `MidpointRounding.ToNegativeInfinity` are not the same thing.